modules/gcp-gke/main.tf

```diff
@@ -165,7 +165,8 @@ resource "google_container_cluster" "primary" {
   }
 
   network_policy {
-    enabled = var.enable_network_policy
+    enabled  = var.enable_network_policy
+    provider = var.enable_network_policy ? "CALICO" : "PROVIDER_UNSPECIFIED"
```

Comment:
This [link](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#nested_network_policy) says `provider` is required only when `network_policy.enabled` is `true`. Please default `provider` to null instead of using `PROVIDER_UNSPECIFIED`.